### PR TITLE
RD2: Update Recurring Donations Batch page description

### PR DESCRIPTION
**Pull request by andrewyu-salesforce**
As a user I want to see accurate help text based on the settings in my org.
# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata
Custom Label:
- stgHelpEnhancedRDBatch

# Deleted Metadata

### CODE REVIEW

**Comment by andrewyu-salesforce** [comment]
**lurch: attach W-041020

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HvJuQAK" target="_blank">W-041020</a>

File: src/labels/CustomLabels.labels [hunk 1]
```diff
@@ -9519,6 +9519,19 @@ Once the process is 100% complete, you can safely leave or refresh this page.</v
         <shortDescription>stgHelpEnableSoftCreditRollups</shortDescription>
         <value>When selected, rolls up Closed/Won Opportunity totals to Contact records based on assigned Opportunity Contact Roles. Soft credit rollups only happen in nightly batches, but you can run them manually from the Rollup Donations Batch settings.</value>
     </labels>
+    <labels>
+        <fullName>stgHelpEnhancedRDBatch</fullName>
+        <categories>Settings</categories>
+        <language>en_US</language>
+        <protected>false</protected>
+        <shortDescription>Enhanced Recurring Donation Batch Intro Help Text</shortDescription>
```
Comment (andrewyu-salesforce) [inline src/labels/CustomLabels.labels:9527]: Screen Shot 2020-04-27 at 10 21 21 AM
@sam-knox Please review the new custom label for the RD2 batch help text

**Comment by andrewyu-salesforce** [comment]
@RadSFDeveloper The code is ready for review

File: src/labels/CustomLabels.labels (hunk 1 above)
Comment (andrewyu-salesforce) [reply src/labels/CustomLabels.labels:9527]: Screen Shot 2020-04-27 at 11 21 38 AM
@sam-knox Updated.

File: src/classes/STG_PanelRDBatch_CTRL.cls [hunk 2]
```diff
@@ -52,6 +52,15 @@ public with sharing class STG_PanelRDBatch_CTRL extends STG_Panel {
     */
     public Boolean isRunningBatch { get; set; }
 
+    /*********************************************************************************************************
+    * Description The Recurring Donation Batch Introduction Text
+    */
+    public String getPageIntro() {
+        return (RD2_EnablementService.isRecurringDonations2Enabled)
```
Comment (RadSFDeveloper) [inline src/classes/STG_PanelRDBatch_CTRL.cls:59]: The `STG_Panel` contains `isRD2Enabled` property that can be used  instead, for example:
 ```
 return isRD2Enabled
    ? System.Label.stgHelpEnhancedRDBatch
    : System.Label.stgHelpRDBatch;
```

File: src/labels/CustomLabels.labels [hunk 3]
```diff
@@ -9519,6 +9519,16 @@ Once the process is 100% complete, you can safely leave or refresh this page.</v
         <shortDescription>stgHelpEnableSoftCreditRollups</shortDescription>
         <value>When selected, rolls up Closed/Won Opportunity totals to Contact records based on assigned Opportunity Contact Roles. Soft credit rollups only happen in nightly batches, but you can run them manually from the Rollup Donations Batch settings.</value>
     </labels>
+    <labels>
+        <fullName>stgHelpEnhancedRDBatch</fullName>
+        <categories>Settings</categories>
+        <language>en_US</language>
+        <protected>false</protected>
```
Comment (RadSFDeveloper) [inline src/labels/CustomLabels.labels:9526]: - Protected should be `true` for every new label.
- Short description can be more descriptive: `Page description displayed when enhanced RD batch is to be run` or similar. @sam-knox

File: src/classes/STG_SettingsManager_TEST.cls [hunk 4]
```diff
@@ -1752,6 +1752,34 @@ private with sharing class STG_SettingsManager_TEST {
...
+        Test.stopTest();
+
+        System.assertEquals(System.Label.stgHelpRDBatch, batchDescription, 'The Legacy RD batch description should be displayed');
+    }
+
+    /***
+    * @description Verifies that when RD2 is enabled, the RD2 batch description should be displayed  
+    */
+    @isTest
+    private static void shouldDispRD2BatchIntroHelpTextWhenRD2IsEnabled() {
```
Comment (RadSFDeveloper) [inline src/classes/STG_SettingsManager_TEST.cls:1773]: `shouldDisplayRD2PageDescriptionWhenRD2IsEnabled()`

File: src/classes/STG_SettingsManager_TEST.cls [hunk 5]
```diff
@@ -1752,6 +1752,34 @@ private with sharing class STG_SettingsManager_TEST {
         system.assertEquals(true, panel.isRunningBatch);
     }
 
+    /***
+    * @description Verifies that when RD2 is not enabled, the legacy RD batch description should be displayed  
+    */
+    @isTest
+    private static void shouldDisplayLegacyRDBatchIntroHelpTextWhenRD2NotEnabled() {
```
Comment (RadSFDeveloper) [inline src/classes/STG_SettingsManager_TEST.cls:1759]: `shouldDisplayLegacyRDPageDescriptionWhenRD2IsNotEnabled()`

File: src/classes/STG_PanelRDBatch_CTRL.cls [hunk 6]
```diff
@@ -52,6 +52,15 @@ public with sharing class STG_PanelRDBatch_CTRL extends STG_Panel {
     */
     public Boolean isRunningBatch { get; set; }
 
+    /*********************************************************************************************************
+    * Description The Recurring Donation Batch Introduction Text
+    */
+    public String getPageIntro() {
```
Comment (RadSFDeveloper) [inline src/classes/STG_PanelRDBatch_CTRL.cls:58]: `getPageIntro()` to `getPageDescription()`?

**Comment by andrewyu-salesforce** [comment]
@RadSFDeveloper Code is update and is ready for the 2nd review

**Pull request merged**